ioc/src/test/java/tw/waterball/survive/ioc/IocContainerTest.java

```diff
@@ -0,0 +1,55 @@
+package tw.waterball.survive.ioc;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import tw.waterball.survive.ioc.exceptions.CycleDependencyException;
+import tw.waterball.survive.ioc.testclasses.acyclicclasses.A;
+import tw.waterball.survive.ioc.testclasses.cyclicclasses.F;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+public class IocContainerTest {
+
+    private static IocContainer iocContainer;
+
+    @BeforeClass
+    public static void init() {
+        iocContainer = new IocContainer();
+    }
+
+    @Test
+    public void scanAcyclicPackage() {
+        iocContainer.scan("tw.waterball.survive.ioc.testclasses.acyclicclasses");
+    }
+
+    @Test
+    public void whenScanCyclicPackage_thenShouldThrowCycleDependencyException() {
+        assertThrows(CycleDependencyException.class,
+                () -> iocContainer.scan("tw.waterball.survive.ioc.testclasses.cyclicclasses"));
+    }
+
+    @Test
+    public void givenAClassType_whenDependencyInjectionInstantiate_thenShouldGetAInstance() {
+        scanAcyclicPackage();
+        Class<A> aClass = A.class;
+        A a = iocContainer.getInstanceByClass(aClass);
+        assertEquals(aClass, a.getClass());
+    }
+
+    @Test
+    public void givenFClassType_whenDependencyInjectionInstantiate_thenShouldThrowDICycleException() {
```

Comment:
I think you mean `thenShouldThrowCycleDependencyException`
to make it consistent with the naming you gave above